### ISSUE TITLE
[v0.5] Questions

### ISSUE THREAD

**Issue by GammaC0de** [pyload/pyload#2584]
@vuolter Two questions:
- When we can see a beta version of v0.5?
- Where we can issue pull requests for v0.5?

**Comment by kvoit** [comment]
Well the Twitter account is as silent as this bug tracker:
https://twitter.com/pyload
Two month ago, code was announced for "in a few weeks", news were "incoming" ... and then, questions have not been answered for the last month.

**Comment by Roburetto** [comment]
Although a couple of strong competitors do apparently exist, the market shift and increased adoption of NAS and miniaturized Arm based devices such as Raspberry, OrangePi etc... gives PyLoad's Python architecture a decisive advantage.

**Comment by wastez** [comment]
I hope it will not end like the first 0.5 Version a few years ago, which had a beta and then .... nothing.
Would be really great if pyload would get back to his old strength.
Hope the new version will come soon.
Thanks for your hard work.

**Comment by Velociraptor85** [comment]
@vuolter 
+1 ^^

**Comment by GammaC0de** [comment]
@vuolter is offline since July :cry:

**Comment by wastez** [comment]
So pyload is dead again??

**Comment by GammaC0de** [comment]
@wastez I cannot tell, hopefully @vuolter will return.
Meanwhile I am taking care of plugin issues.

**Comment by Jonn3y** [comment]
Pyload has to be alive again! :(

**Comment by GammaC0de** [comment]
@Jonn3y I've got a message  from @vuolter few days ago.
He is now back and he should resume his work soon (that's what he promised)

**Comment by wastez** [comment]
Are there any news?
Is voulter again working on it?

**Comment by vuolter** [comment]
@wastez I just have a few posts to read before do it...
![notifications](https://cloud.githubusercontent.com/assets/1221092/19415697/5bf4c41a-9378-11e6-942e-0930e2c52275.png)
:weary:

**Comment by BKPepe** [comment]
Glad to see that @vuolter  is back and also alive!

**Comment by Velociraptor85** [comment]
Happy to see @vuolter back here :+1:

**Comment by godfuture** [comment]
Welcome back @vuolter

**Comment by vuolter** [comment]
Thank you all.

**Comment by GammaC0de** [comment]
@vuolter You should post something on Twitter..

**Comment by Ryan747800B** [comment]
Welcome back @vuolter !

**Comment by m33cr0b** [comment]
I cant wait to see this beta, since multi hoster support is broken, pyload has just become worthless to me!

**Comment by godfuture** [comment]
Yup, pyload is too nice to fall like that...I will try to help the best way I can.

**Comment by Burnout5151** [comment]
Great to see a discussion about the future development.
I think the way PyLoad is currently developed is not feasible:

- Addons are updated constantly (rolling release)
- The core is at a fixed revising (not rolling release) and very old (0.4.9)
- No patches to the core are accepted
- Development of the next big version is  being done by a single developer, offline

I think "0.5" should be developed online in this git-repo
I think PyLoad needs a model how to be further developed until "0.5" arrives.
And so I guess the first question is:
Should there be a full rolling-release with a "stable" and a "testing"-branch?
Or should there only be versioned packages?
Or the approach: Having a stable branch and just tagging at defined version numbers?

I'm willing to help with development or releases in my spare time.

**Comment by kvoit** [comment]
I think I remember statements from @vuolter that stable/testing branches are planned with v0.5. Looking at the pyload tweets, v0.5 is about to arrive, so I don't think it makes sense to start big discussions about models "until v0.5 arrives".

Regarding your points:
1. I think a rolling release of plugins is necessary. Hosters change their interface and this needs immediate reaction. There is no room at all for chains with staging/testing/unstable and such.
2./3. Hopefully, this changes with v0.5
4. I agree that this is unfortunate. On the one hand, of course this is @vuolter project, he has dedicated enormous amounts of time and he can do whatever he thinks is best and the only thing we have the right to do is to be grateful. On the other hand, though, for a lot of us, pyload is "critical infrastructure" and it is running problematically and has been for a while. This is made much worse by the fact that there is no real viable alternative for headless solutions. IN addition, other people, namely @GammaC0de have ALSO dedicated a lot of time, which deserves some perspective.

In my opinion, pyload is a great piece of software and it deserves some more patience. From all we know, there is a great new version coming up. However, if the project stays in such a closed, stalled state, in the medium term something has to be done about alternatives, since as I said, it is critical infrastructure for many.  The month between July and October have been quite scary in that regard. I hope @vuolter  takes this as nothing but a compliment.

If it comes to the necessity of alternatives, also offer my help. Right now, I don't think it makes sense to either get deeper into code that is likely to be replaced soon, or start projects that are hopefully likely to be obsolete soon with v0.5.

**Comment by Burnout5151** [comment]
Greetings, I didn't want to denigrate/bad mouth this project or any developer. I'm sorry if it came across that way. 

That being said, it seems like I should first make a point _why_ I think there needs to be a concept for continuing development before "0.5". _How_ to develop can be discussed if we reach a consensus.

So here's my point: While it ia true that work on 0.4.10 would become obsolete with 0.5, I think that 0.5 won't be coming soon. I guess it has been under development for about 1.5 years(?) and according to the [milestones page](https://github.com/pyload/pyload/milestones) it is 65% done. If you are not a very seasoned developer, the last 10% are much slower than the first 10%. These milestones don't seem to consider the rewrite or the new hoster plugins. So my estimate is that 0.5 will be here in about ~2 years.
Yes I did read the tweet about updating 0.5 source code in July(?) but I don't think anything happened.
This is not to rip on anybody personally. This is just my reasoning why we need an active development instead of waiting for 0.5.
I don't think new developers will join either. I was very frustrated when I opened a PullRequest (#2473) for an easy bugfix and it was turned down because "no fixes to core". Then I found that somebody else had fixed the same problem half a year earlier and was turned down for the same reason.
Or look at Extractarchive issue #2168: It took half a year from "70%" until someone else fixed it, but to me it looks like it will be shipped in 0.5. Meanwhile people are creating their own Shellscripts to add this functionality.
Now i stumbled upon a bug with non-ascii characters, which brought me back to this dev-board.
I found somebody else already has the same problem...https://github.com/pyload/pyload/issues/2286#issuecomment-258676317 with half a dozen linked issues. Some people seem to have developed patches that work for the characters they need, but since there is no central place where development happens, no patch fixes everything.

If the software doesn't work and there is no hope in fixing it, I think not only the userbase will dwindle but also no new developers will join.

Please let me know what you think.

Greetings
Burnout

**Comment by kvoit** [comment]
I totally agreed and agree with you that that the current situation is a mess. A few things seem to have improved after I recently pulled again (after not pulling for month because I got more problems than solved at that time), e.g., extracting seems to work again, an https proxy isn't forwarded to http all the time anymore and similar things. I still have some hacked fork that I always merge to make "C'nL directly to queue" working.
Anyway. In my head, I am skipping the last few month since July, since @vuolter explained he had other problems but is back now, so the "few weeks" are running again. I wouldn't give too much on the milestone counter, that doesn't say anything for most project, because it is not properly maintained.

I'd say, give it until end of the year. That would be, about a year after heavy stuff like the ExtractArchive stuff when down the drain. Then, we should consider if a fork or a development from scratch makes sense.

**Comment by Burnout5151** [comment]
Well, I guess we have similar experience then. I'm still not 100% convinced that we should not at least merge existing CRs, but I can understand that a CR is quickly merged, but hard to debug if merged without care.
So, okay let's wait till the end of the year and hope for the best.
And if the situation is still the same by then, I hope you are still around, to discuss the situation then.

**Comment by godfuture** [comment]
No matter when and what will be released with 0.5, the issue is that some "core parts" of pyload are updated with rolling release. It would be easier, if certain modules like extractArchive would move to core, such that releases are more stable. For hoster plugins, it still makes sense to update constantly.

The big question is which modules could possibly belong to a "pyload core", I guess...

**Comment by Velociraptor85** [comment]
I think the main Problem is the communication :)
Maybe someone could setup an Rocketchat or Slack Room? Preferably something which could be linked somewhere one Github too :)

Have a nice day ! ;)

**Comment by kvoit** [comment]
@godfuture Sounds pretty easy: Things with "outside contact" that need to change with "outside change" will need to update immediately. Everything else can have a release cycle, with the usual emergency plan for serious security problems.

@Velociraptor85 I agree that the project has a communication problem. But I think the problem is that the only developer rarely communicates, not that people who are right now not actively developing cannot communicate properly.

**Comment by Jonn3y** [comment]
I hope it will be a nice christmas present

**Comment by Burnout5151** [comment]
@Velociraptor85 @kvoit I think for the regular development (discussing PRs) but in case there is a big strategy meeting maybe there should be an IRC-Conference or a Mailinglist (we might be in very different timezones). But currently I DO have the feeling that things are getting better.
Let's hope for Christmas :-)

**Comment by Burnout5151** [comment]
Uh, nice, I just saw a branch "1.0.0" was opened 2 days ago! :-)

**Comment by Burnout5151** [comment]
@vuolter could you give a short info on the 1.0.0 branch?
Is it ready to be debugged/extended/tested/used?
Tanks you for your work :-)

**Comment by vuolter** [comment]
@Burnout5151 In these holidays I hope to complete the conversion to python3 of the old 0.5.0 branch developed by @RaNaN (nowadays named 1.0.0), and merge with my local master branch, so I recommend to not open PR for the moment. :s

**Comment by Burnout5151** [comment]
Sounds ambitious,I wish you good progress and still enough time for the holidays. :-)
I will then refrain from developing for the time being.
Greetings to everyone and happy holidays.

**Comment by Velociraptor85** [comment]
@vuolter so how is the effort going? Any Progress with your Plan? Its again 2 weeks after your last post would be happy to see some report :) maybe even some call for help if needed

cu

**Comment by vuolter** [comment]
@Velociraptor85 Working on 1.0.0 as written. :s

**Comment by BKPepe** [comment]
ETA?

**Comment by vuolter** [comment]
@BKPepe Not now. At this point I do not want to jinx! :p

**Comment by Velociraptor85** [comment]
@vuolter Hej Walter any news ? Its again 2 Weeks gone, on which part are you working now? Is there any progress report for us =)?

cu

**Comment by vuolter** [comment]
Actually, I took another break from the project, but I'll try to complete the core in the next week-end. With the new core up and ready I just have to fix the old libs to make it working on Python 3. :)
All the other new modules will be merged progressively.

**Comment by Frozen-byte** [comment]
I'm super excited to see that 0.5 brings!

**Comment by Mister-XY** [comment]
This weekend must be the next weekend or?

**Comment by Velociraptor85** [comment]
well we hope that @vuolter did what he said =) maybe this week there will be a release =)

**Comment by Mister-XY** [comment]
Hm, nothing happened.

**Comment by Velociraptor85** [comment]
@Mister-XY well thats not completely correct https://github.com/pyload/pyload/commits/1.0.0 @vuolter did some commits on Friday 👍

**Comment by Jonn3y** [comment]
Is 1.0.0 usable?!

**Comment by vuolter** [comment]
@Jonn3y Soon... I've just completed today the porting with most of the libs. It took me two days of full-work. :(

**Comment by sosnet** [comment]
Thanks a lot @vuolter! And it's done when it's done :+1:

**Comment by Velociraptor85** [comment]
@vuolter Great news ! :) 👍 should we already test the 1.0.0 branch and report issues ?

**Comment by darky112** [comment]
How I can install the 1.0.0 Version
I always get this error
```
Traceback (most recent call last):
  File "setup.py", line 9, in <module>
    from builtins import PACKDIR, dict
ImportError: cannot import name 'PACKDIR'
```
When I am running 
`
python setup.py 
`

**Comment by wastez** [comment]
It's not working until now.

**Comment by darky112** [comment]
@wastez so bad. :(

**Comment by Velociraptor85** [comment]
Hej @vuolter so I saw that there is progress on 1.0.0 and the stable branch anything we could do? Or what are the next steps to take? Would love some comment :)

**Comment by snickers2k** [comment]
i would be happy to beta-test 0.5 as soon as there is something installable :D

**Comment by kvoit** [comment]
Given the state of the 0.49 version, I think everybody would be happy to beta-test :)

**Comment by Burnout5151** [comment]
Hi,
I was gone for a while, can someboday bring me up to speed?
There is a stable branch, but in #2794 I can read that it is not the default branch?
Is it not the default to push to or to clone from?
Are CRs accepted to either?

**Comment by Velociraptor85** [comment]
HI @Burnout5151 stable branch is the current working branch which is still maintained by @GammaC0de whom keeps it running its the "old" running 0.4.* branch which works and you can clone and push to ... the master branch is the working branch for 0.5 release which @vuolter is working on as far as I know he is trying to separate it to more branches/modules and than have those maintained
For now 0.5 is not working, that's at least my last info, and I am not sure how much time @vuolter has currently or if there is an ETA or some roadmap 

hopefully I could help ... for more info I guess you could also write to gitter or irc ;)

**Comment by Burnout5151** [comment]
Thank you for the the explanation.

Also for the reference to the other means of communication. I will check them out.

**Comment by Jonn3y** [comment]
Hey, any new stuff here?

**Comment by Floflobel** [comment]
UP ?

**Comment by sNiXx** [comment]
Any news on current progress?

**Comment by GammaC0de** [comment]
v0.5 is not planned to be realeased in the near future due to @vuolter is very busy.

I do plan to release https://github.com/GammaC0de/pyload/tree/captcha-interactive as v0.4.10 with interactive captcha support.

**Comment by sNiXx** [comment]
Any news regarding v0.4.10 or v0.5? It has been over a year since vuolter said it would be here soon. At some point it might be better to fork this project, as suggested a few years ago? It is too bad that there is no definitive information - I think it would really help improve this fantastic project.
Either way, thank you for all your work @GammaC0de!

**Comment by hacki11** [comment]
Perhaps making small feature increments and smaller releases would be more suitable here and would give more benefit for the user. that should be no critic but it seems the planned features are too big to get released with the given resources. (few years for an increment 0.4 => 0.5).

**Comment by GammaC0de** [comment]
@StefftheEmperor2
> Where?

https://github.com/StefftheEmperor2/pyload/tree/dev/fixes

**Comment by StefftheEmperor2** [comment]
@GammaC0de thx for moving that post to the right position. The point is, the company I work for is going to be closed on may the 1st and we don't get any new tasks assigned. So I have a LOT of slack time. So I thought this was a good opportunity to start learning python and help with this project. In fact, I use pyload for a very long time. I'd like to get the project tasks coordinated with you, to push the development of v.5 to the next level of that great software.

**Comment by GammaC0de** [comment]
HI @StefftheEmperor2 ,
The lead developer of v0.5 is @vuolter .
He can tell us what is left to be done before v0.5 can be released.
Best way to contact him is by gitter (https://gitter.im/pyload/pyload).
Please note that he is very busy nowadays.

Of Course I would do my best to help you.

P.S.
IRC probably closed, nobody hang out there anymore.

**Comment by StefftheEmperor2** [comment]
Yes I know he's very busy. But pyload is free software. The development shouldn't depend on a single person but should thrive by its community and the people who are willing to contribute.

**Comment by GammaC0de** [comment]
> The development shouldn't depend on a single person but should thrive by its community and the people who are willing to contribute.

Yes you are right about that.

I had sent @vuolter a message via Gitter and asked him what has to be done before release of v0.5.
Lets wait a few days to his answer.

**Comment by StefftheEmperor2** [comment]
Any news on this @GammaC0de @vuolter

**Comment by StefftheEmperor2** [comment]
Today I switched the status polling to pushing via websockets

**Comment by GammaC0de** [comment]
> Today I switched the status polling to pushing via websockets 

Nice

**Comment by StefftheEmperor2** [comment]
Today I started writing a Firefox extension for handling interactive captchas

**Comment by GammaC0de** [comment]
> Today I started writing a Firefox extension for handling interactive captchas

If you find a way to fool an iframed web page window top that would be a great contribution

**Comment by StefftheEmperor2** [comment]
My extension works now ;-)

**Comment by StefftheEmperor2** [comment]
![Screenshot_20200407-213708](https://user-images.githubusercontent.com/55734627/78718049-d7d28080-7921-11ea-87b3-475bf26e57ec.jpg)

**Comment by GammaC0de** [comment]
@StefftheEmperor2 These crazy days , as many people are at home, makes a lot of feature requests and bug reports so I had no time to try 0.5.

Also I'm having issues to install lxml under windows :(
